manifests/register.pp

```diff
@@ -0,0 +1,24 @@
+# @summary Register the host as Foreman
+# @api private
+class foreman::register (
+  Stdlib::Fqdn $foreman_host_name = $facts['networking']['fqdn'],
+) {
+  ensure_resource('foreman_host', $foreman_host_name, {
+      ensure          => present,
+      base_url        => $foreman::foreman_url,
+      consumer_key    => $foreman::oauth_consumer_key,
+      consumer_secret => $foreman::oauth_consumer_secret,
+      effective_user  => $foreman::oauth_effective_user,
+      ssl_ca          => $foreman::server_ssl_ca,
```

Comment:
Oh doh, I am an idiot. This CA file is going to be different from the context of Foreman and from the context of the smart-proxy. On Foreman, this path may be the Puppet CA or in Katello's case a path in `/etc/foreman` whereas from the smart-proxy perspective, the CA file used to communicate with Foreman is in `/etc/foreman-proxy`. So we cannot rely on `ensure_resource` giving us the same resource across both modules due to this parameter.

I think then we have two options off the top of my head:

 1) Add a parameter for the hostname and then make the titles unique e.g. `foreman-foreman.example.com` and `foreman-proxy-foreman.example.com`
 2) A class that is shared between puppet-foreman and puppet-foreman_proxy to ensure creation of the host object since puppet-foreman_proxy requires puppet-foreman